app/models/evaluation_form.rb

```diff
@@ -17,6 +17,7 @@
 #
 class EvaluationForm < ApplicationRecord
   belongs_to :challenge
+  has_many :evaluation_criteria, dependent: nil
```

Comment:
this should be dependent: :destroy so the associated criteria are destroyed when the form is destroyed
```suggestion
  has_many :evaluation_criteria, dependent: :destroy
```